spec/controllers/assignments_controller_spec.rb

```diff
@@ -166,6 +170,19 @@
           .to redirect_to roster_assignments_path(roster, date: Time.zone.today)
       end
 
+      context 'when the end date is before the start date' do
+        before { params[:end_date] = Date.yesterday.to_fs(:db) }
+
+        it 'warns the end date is before the start date' do
+          submit
+          expect(flash[:errors]).not_to be_empty
+        end
+
```

Comment:
```suggestion
```
We generally don't test flash messages at this level. Flash messages fall under "page content" which is more of a fit for system specs (you'll only see other examples here because this is an old controller spec).